esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -551,6 +645,75 @@ static inline bool isPublishedQuery(EsdlMethodImplType implType)
     return (implType==EsdlMethodImplRoxie || implType==EsdlMethodImplWsEcl);
 }
 
+void EsdlServiceImpl::handleFeatureAuthorization(IEspContext& context, IEsdlDefMethod& esdlDef)
+{
+    auto method = esdlDef.queryName();
+    auto bindingMethodAccessMap = m_bindingAccessMaps.getValue(method);
+    MethodAccessMap* accessMap = nullptr;
+#ifdef DEBUG
+    const char* accessMapSource;
+#endif
+
+    if (bindingMethodAccessMap && bindingMethodAccessMap->get())
+    {
+        // The existence of an access map from the binding gives the binding precedence
+        // over the ESDL definition.
+        accessMap = bindingMethodAccessMap->get();
+#ifdef DEBUG
+        accessMapSource = "binding";
+#endif
+    }
+    else
+    {
+        // The absence of an access map from the binding defers the security specification
+        // to the ESDL definition.
+        accessMap = &const_cast<MethodAccessMap&>(esdlDef.queryAccessMap());
```

Comment:
Often demonstrates the the use of const was wrong, either in the source or destination, if you have to cast it away.

Reply:
I agree completely. IEsdlDefMethod returns const and IEspContext requires non-const. If you check original line 575 you'll see that this cast was necessary before my changes.

Reply:
Understood.  Just making sure it was thought about.  When const_cast is appropriate is when something is meant to be const for just about everyone, but you are well aware you are the one case that wants to change the contents.

Reply:
Saw this going by, const_cast is nasty.. I suspect the original code on line 575 was abusing it too.

>..the one case that wants to change the contents.

I don't think anything is changing the content.. 
Looking at how 'accessMap' is being used here, it looks like 'validateFeaturesAccess' is the only method that is requiring it to be non-const, and having had a look at it, it looks like the map it accepts as an argument should be const. i.e. validateFeaturesAccess is not const correct (it does not alter the object that's passed into it).

So I think the correct fix is to make validateFeaturesAccess accept a const and then you wouldn't need the const_cast kludge.